nix/mobile/android/targets/release-android.nix

```diff
@@ -0,0 +1,129 @@
+  buildType' = if (build-type == "release" || build-type == "nightly") then "release" else "pr";
+  generatedApkPath = "android/app/build/outputs/apk/${buildType'}/app-${buildType'}.apk";
+  outApkName = "app.apk";
+
+in stdenv.mkDerivation {
+  inherit name;
+  src =
+    let path = ./../../../..;
+    in builtins.path { # We use builtins.path so that we can name the resulting derivation, otherwise the name would be taken from the checkout directory, which is outside of our control
+      inherit path;
+      name = "status-react-${name}-source";
+      filter =
+        # Keep this filter as restrictive as possible in order to avoid unnecessary rebuilds and limit closure size
+        mkFilter {
+          dirRootsToInclude = [ 
+            "mobile_files"
+            "modules/react-native-status"
+            "packager"
+            "resources"
+          ];
+          dirsToExclude = [ ".git" ".svn" "CVS" ".hg" ".gradle" "build" "intermediates" "libs" "obj" ];
+          filesToInclude = [ envFileName "STATUS_GO_VERSION" "VERSION" ];
+          root = path;
+        };
+    };
+  nativeBuildInputs = [ bash gradle ] ++ lib.optionals stdenv.isDarwin [ file gnumake watchman ];
+  buildInputs = [ nodejs openjdk ];
+  phases = [ "unpackPhase" "patchPhase" "buildPhase" "installPhase" ];
+  postUnpack = ''
+    mkdir -p ${gradleHome}
+
+    ${if keystore-file != "" then "cp -a --no-preserve=ownership ${keystore-file} ${gradleHome}/; export STATUS_RELEASE_STORE_FILE=${gradleHome}/$(basename ${keystore-file})" else ""}
+
+    # Ensure we have the right .env file
+    cp -f $sourceRoot/${envFileName} $sourceRoot/.env
+
+    # Copy index.*.js input file
+    cp -a --no-preserve=ownership ${prod-build}/index*.js $sourceRoot/
+
+    # Copy android/ directory
+    cp -a --no-preserve=ownership ${sourceProjectDir}/android/ $sourceRoot/
+    chmod u+w $sourceRoot/android
+    chmod u+w $sourceRoot/android/app
+    chmod -R u+w $sourceRoot/android/.gradle
+    mkdir $sourceRoot/android/build && chmod -R u+w $sourceRoot/android/build
+
+    # Copy node_modules/ directory
+    #ln -sf ${sourceProjectDir}/node_modules/
+
+    # Make android/build directories writable under node_modules
+    cp -a --no-preserve=ownership ${sourceProjectDir}/node_modules/ $sourceRoot/
+    chmod -R u+w $sourceRoot/node_modules/react-native/
+    for d in `ls $sourceRoot/node_modules/react-native-*/android/build -d1`; do
+      chmod -R u+w $d
+    done
+    for d in `ls $sourceRoot/node_modules/react-native-*/android -d1`; do
+      chmod u+w $d
+    done
+    chmod u+w $sourceRoot/node_modules/realm/android
+  '';
+  patchPhase = ''
+    prevSet=$-
+    set -e
+
+    substituteInPlace android/gradlew \
+      --replace 'exec gradle' "exec gradle -S -Dmaven.repo.local='${localMavenRepo}' --offline ${gradle-opts}"
+    substituteInPlace android/app/build.gradle \
+      --replace 'versionCode 9999' 'versionCode ${build-number}'
```

Comment:
Oh, so we set it to `9999` earlier to avoid re-building stuff, but now in the release build we change it to the actual value. Messy, but I get it. Maybe a better solution would be to implement something in `build.gradle` to cover this, rather than do this kind of patching?

Reply:
Added a property to gradle.properties:
```
versionCode=9999
```

This property gets replaced in a release-android build.